heat/core/indexing.py

```diff
@@ -69,18 +70,17 @@ def nonzero(x: DNDarray) -> DNDarray:
 
     if x.ndim == 1:
         lcl_nonzero = lcl_nonzero.squeeze(dim=1)
-    for g in range(len(gout) - 1, -1, -1):
-        if gout[g] == 1:
-            del gout[g]
-
-    return DNDarray(
-        lcl_nonzero,
-        gshape=tuple(gout),
-        dtype=types.canonical_heat_type(lcl_nonzero.dtype),
-        split=is_split,
-        device=x.device,
-        comm=x.comm,
-        balanced=False,
+
+    return tuple(
+        DNDarray(
+            lcl_nonzero,
+            gshape=tuple(gout),
+            dtype=types.canonical_heat_type(lcl_nonzero.dtype),
+            split=is_split,
+            device=x.device,
+            comm=x.comm,
+            balanced=False,
+        )
     )
```

Comment:
This would return a tuple with 1 element (a 2-dimensional DNDarray). What we need is a tuple of 1-D DNDarrays, namely the columns of the original `lcl_nonzero` (output of `torch.nonzero` operation)

Reply:
No. It does work the way we want.
Let us say we have a list of lists in python (a 2-d array basically). When we do, `tuple(arr)`, we get a tuple of 1-D arrays only not a tuple of 2-D array.

Reply:
Better yet, let me show some demo output of this implementation.

Reply:
Consider this code snippet.
![image](https://user-images.githubusercontent.com/73862377/159730309-b62247ea-781e-40c6-aaf5-db94e03a045a.png)

Reply:
The output is something like this.
![image](https://user-images.githubusercontent.com/73862377/159730498-a42c6c6a-be85-458f-8de4-741967e9c251.png)